wireapp/kalium
# fix: wrong failType for MLS federation errors [WPB-9087]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9087" title="WPB-9087" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9087</a>  [Android] Learn more links in conversation error message not displayed anymore
  </td></table>
  <br />
 






----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

"Learn more" links in conversation error message are not displayed anymore for MLS errors when creating a conversation and adding members.

### Causes (Optional)

Wrong `failType` for MLS federation errors.

### Solutions

Set the right `failType`, also use correct `failType` for non-retryable errors.

### Testing

#### Test Coverage (Optional)


#### How to Test

STR:
- have users A and B from backend A
- have user C from backend B
- login as user A
- start group creation flow, select user B and C
- take user Cs backend offline
- continue creating the group conversation
- Error message is displayed about that user C could not be added

Expected:
The error message should have a link “Learn more” which forwards the user to an FAQ page

Actual:
The “Learn more” link is missing.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
